File: airesources/Go/RandomBot.go
package main

import (
	"hlt"

	"math/rand"
)

func main () {
	conn, gameMap := hlt.NewConnection("RandomBot")
	for {
		var moves hlt.MoveSet
		gameMap = conn.GetFrame()
		for y := 0; y < gameMap.Height; y++ {
			for x := 0; x < gameMap.Width; x++ {
				loc := hlt.NewLocation(x,y)
				if gameMap.GetSite(loc, hlt.STILL).Owner == conn.PlayerTag {
					moves = append(moves, hlt.Move {
						Location: loc,
						Direction: hlt.Direction(rand.Int() % 5),

					})
				}
			}
		}
		conn.SendFrame(moves)

	}
}

File: airesources/C/RandomBot.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#include "hlt.h"

#define BOT_NAME "RandomCBot"

int main(void) {

    GAME game;
    int x, y, direction;

    srand(time(NULL));

    game = GetInit();
    SendInit(BOT_NAME);

    while (1) {

        GetFrame(game);

        for (x = 0 ; x < game.width ; x++) {
            for (y = 0 ; y < game.height ; y++) {
                if (game.owner[x][y] == game.playertag) {
                    direction = rand() % 5;
                    SetMove(game, x, y, direction);
                }
            }
        }

        SendFrame(game);
    }

    return 0;
}
